osv/repos.py

```diff
@@ -115,23 +117,14 @@ class RepoInaccessibleError(Exception):
 def clone(git_url, checkout_dir, git_callbacks=None, blobless=False):
   """Perform a clone."""
   try:
-    # Use 'git' CLI here as it's much faster than libgit2's clone.
-    env = _set_git_callback_env(git_callbacks)
-    cmd = ['git', 'clone']
-    if blobless:
-      cmd.append('--filter=blob:none')
-    cmd.extend([_git_mirror(git_url), checkout_dir])
-    subprocess.run(cmd, env=env, capture_output=True, check=True)
+    os.makedirs(checkout_dir, exist_ok=True)
+    cmd = ['curl', f"{GITTER_HOST}/getgit?url="+ _git_mirror(git_url), "-o", checkout_dir+".zst", "-s"]
+    subprocess.run(cmd, check=True)
+    cmd = ['tar', '-xf', checkout_dir+".zst", "-C", checkout_dir]
+    subprocess.run(cmd, check=True)
     return pygit2.Repository(checkout_dir)
   except subprocess.CalledProcessError as e:
-    stderr = e.stderr.decode(errors='ignore')
-    if ('could not read Username' in stderr or
-        ('fatal: repository' in stderr and 'not found' in stderr) or
-        'Authentication failed' in stderr):
-      # Git is asking for username/password, the repository doesn't exist, or
-      # authentication failed.
-      raise RepoInaccessibleError() from e
```

Comment:
You need to make sure you retain similar logic to this in the gitter and pass it through here - it's important for malicious packages where the repo was deleted

Reply:
Done.